### PR TITLE
walletConnect added for reward claim

### PR DESCRIPTION
**Pull request by Raj6939**
- changed made in myWallet.js( configuration for walletConnect due to upgrade in walletConnect) ref [here](https://github.com/WalletConnect/web3modal/releases/tag/2.2.0)
- replaced wait function by listening contract events after calling claiming (withdraw method).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,30 +5,36 @@
...
+    "@wagmi/core": "^0.10.1",
+    "@web3modal/ethereum": "^2.2.0",
+    "@web3modal/html": "^2.2.0",
     "auth0-js": "^9.16.2",
     "axios": "^0.21.1",
     "bip39": "^3.0.4",
     "bootstrap-vue": "^2.0.0-rc.11",
     "browserify": "^17.0.0",
     "bs58": "^5.0.0",
+    "core-js": "^3.8.3",
```
Comment (Vishwas1) [inline package.json:32]: why do we need core-js and crypto-browserify libraries? are we using it anywhere?

File: package.json [hunk 2]
```diff
@@ -5,30 +5,36 @@
...
+    "crypto-browserify": "^3.12.0",
     "dayjs": "^1.10.6",
     "dotenv": "^16.0.0",
     "elliptic": "^6.5.4",
+    "ethers": "^5",
```
Comment (Vishwas1) [inline package.json:37]: we already have web3 js , do we need etherjs too ? where are you using it?

File: package.json [hunk 3]
```diff
@@ -37,6 +43,7 @@
     "marked": "^3.0.8",
     "multicoin-address-validator": "^0.5.10",
     "node-fetch": "^2.6.0",
+    "path-browserify": "^1.0.1",
```
Comment (Vishwas1) [inline package.json:46]: whats the use of this new package?

File: package.json (hunk 1 above)
Comment (Raj6939) [reply package.json:32]: yes we are using crypto package in action.vue page to sign every action. But now we shifted to yarn and we cannot use crypto package directly in yarn so we need wrapper on that which is path-browserify. Take a look here in vue.config.js https://github.com/hypersign-protocol/hyperfyre-frontend/pull/1915/files#diff-1d62a367f02f5c91000e35c82b0086fae3246eca3942f9034c058b16a3780ec2

File: package.json (hunk 2 above)
Comment (Raj6939) [reply package.json:37]: yes we need since wagmi/core uses internally ethers.js

File: package.json (hunk 3 above)
Comment (Raj6939) [reply package.json:46]: yes we are using crypto package in action.vue page to sign every action. But now we shifted to yarn and we cannot use crypto package directly in yarn so we need wrapper on that which is path-browserify. Take a look here in vue.config.js https://github.com/hypersign-protocol/hyperfyre-frontend/pull/1915/files#diff-1d62a367f02f5c91000e35c82b0086fae3246eca3942f9034c058b16a3780ec2

File: package.json (hunk 1 above)
Comment (Raj6939) [reply package.json:32]: about core-js it provides polyfill for running modern code to run in old browsers.I think i have installed that while i was getting optional chaining error. it suggests me in error to install core-js.

File: package.json (hunk 2 above)
Comment (Raj6939) [reply package.json:37]: Sorry i am not using it. I have installed it way back while developing the feature.

File: package.json (hunk 3 above)
Comment (Raj6939) [reply package.json]: removed path-browserify pakcages and tested. It worked since i was doing configuration in wrong way in vue.config.js file

**Pull request merged**